### DISCUSSION TITLE
How to destroy cloud-init drives after vm creation is complete.

### DISCUSSION THREAD

**Discussion by Uncurlhalo** [bpg/terraform-provider-proxmox#1542] [Q&A]
I'm trying to understand how to approach destroying all the cloud-init drives that are created when i stand up a large number of VM's. They aren't very large but it seems like wasted storage. What is the recommended way to do this?

**Comment by bpg-dev** [comment]
Hey @Uncurlhalo 👋🏼 

There's no 'recommended' way to do this, since those cloud-init drives are still referenced in the VM configs in PVE, not just in the Terraform state. You can try manually removing `cicustom` from the VM configs in `/etc/pve/qemu-server/<VM ID>.conf` and then from the TF state, then manually delete the disks. But if you have a lot of VMs, I’m not sure the hassle is worth it.

**Reply by Uncurlhalo** [reply]
Thanks for the reply. yeah its a bit of a headache and adding them all up it's less than 250MB so honestly it's not worth the effort at this point. If it ever because valuable enough to me I might explore a way to try and automate the process of unassociating them and then updating terraform state.
